index/src/index.rs

```diff
@@ -481,18 +489,38 @@ impl Index {
         }
     }
 
-    fn split_text_into_words(text: &str) -> Vec<String> {
-        // Define a regular expression to remove all non-alphanumeric characters except spaces
-        let re = Regex::new(r"[^\w\s]").unwrap();
-
-        // Remove newlines and special characters from the text
-        let cleaned_text = re.replace_all(text, "");
+    fn create_page_index(path: &mut PathBuf, body: &str) -> Result<()> {
+        // Create reversed index map
+        let pindex: PageIndex = Self::split_text_into_words(body)?;
+        path.set_extension("pageindex");
```

Comment:
I think it would be cleaner if this function took an immutable reference, which would also allow to get rid of the clones above.

```suggestion    
    fn create_page_index(path: &Path, body: &str) -> Result<()> {
        // Create reversed index map
        let pindex: PageIndex = Self::split_text_into_words(body)?;
        let path = path.with_extension(".pageindex");
```